types/properties-reader/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for properties-reader 0.0
+// Project: https://github.com/steveukx/properties
+// Definitions by: Zlatko Andonovski <https://github.com/Goldsmith42/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace PropertiesReader {
+    type Value = string|number|boolean;
+
+    interface Reader {
+        get(propertyName: string): Value|null;
+        getRaw(propertyName: string): string|null;
+        path(): Object;
+        append(path: string): Reader;
+        read(properties: string): Reader;
+        set(propertyName: string, value: Value): Reader;
+        length: number;
+        each(iterator: (key: string, value: Value) => void, scope?: Object): Reader;
```

Comment:
It's the context to which the iterator function is bound. What `each` does is that it calls `iterator` for each property. If you use `this` inside iterator, it'll refer to the object provided with `scope`, otherwise it'll be the reader object itself.